src/beamlime/applications/handlers.py

```diff
@@ -11,10 +13,21 @@
 from beamlime.logging import BeamlimeLogger
 
 from ..stateless_workflow import StatelessWorkflow, WorkflowResult
+from ._nexus_helpers import combine_store_and_structure, merge_message_into_store
 from .base import HandlerInterface
-from .daemons import DetectorDataReceived, RunStart
+from .daemons import (
+    ChopperDataReceived,
+    DetectorDataReceived,
+    LogDataReceived,
+    RunStart,
+)
 
 Events = NewType("Events", list[sc.DataArray])
+MergeMessageEveryNth = NewType("MergeMessageEveryNth", Number)
+'''Every n:th message the data assembler receives, the data reduction is run'''
```

Comment:
Not so important but, how about keeping the names similar with `MergeMessageTimeInterval`...?
```suggestion
MergeMessageCountInterval = NewType("MergeMessageCountInterval", Number)
'''Every MergeMessageCountInterval-th message the data assembler receives, the data reduction is run'''
```

Reply:
I think this was a very good suggestion. I'll add a separate pr for those since this is already merged now.